Rebus.AdoNet.Tests/SagaPersisterTests.cs

```diff
@@ -318,31 +399,127 @@ public void WhenIgnoringNullProperties_DoesNotSaveNullPropertiesOnInsert()
 		[Test]
 		public void CanCreateSagaTablesAutomatically()
 		{
-			// arrange
+			var tableNames = GetTableNames();
+			Assert.That(tableNames, Contains.Item(SagaTableName), "Missing main saga table?!");
 
-			// act
-			CreatePersister().EnsureTablesAreCreated();
+			var schema = GetColumnSchemaFor(SagaTableName); //< (item1: COLUMN_NAME, item2: DATA_TYPE, [...?])
+			var indexes = GetIndexesFor(SagaTableName);
+			Assert.Multiple(() =>
+			{
+				Assert.That(Dialect.GetDbTypeFor(schema.First(x => x.Item1 == "id").Item2), Is.EqualTo(DbType.Guid), "#0.0");
+				Assert.That(Dialect.GetDbTypeFor(schema.First(x => x.Item1 == "revision").Item2), Is.EqualTo(DbType.Int32), "#0.1");
+				Assert.That(schema.First(x => x.Item1 == "saga_type").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#0.2");
+				Assert.That(schema.First(x => x.Item1 == "data").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#0.3");
+				Assert.That(indexes, Contains.Item($"ix_{SagaTableName}_id_saga_type"), "#0.4");
+			});
+
+			if (_persister is AdoNetSagaPersisterLegacy) //< NOTE: DATA_TYPE for PostgresSQL arrays are _text instead of text[].
+			{
+				Assert.Multiple(() =>
+				{
+					var sagaIndexSchema = GetColumnSchemaFor(SagaIndexTableName);
+					var sagaIndexIndexes = GetIndexesFor(SagaIndexTableName);
+					var isArray = _features.HasFlag(Feature.Arrays);
+					Assert.That(tableNames, Contains.Item(SagaIndexTableName), "Missing saga indexes table?!");
+					Assert.That(Dialect.GetDbTypeFor(sagaIndexSchema.First(x => x.Item1 == "saga_id").Item2), Is.EqualTo(DbType.Guid), "#1.0");
+					Assert.That(sagaIndexSchema.First(x => x.Item1 == "key").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#1.1");
+					Assert.That(sagaIndexSchema.First(x => x.Item1 == "value").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#1.2");
+					Assert.That(sagaIndexSchema.First(x => x.Item1 == "values").Item2, Is.EqualTo(isArray ? "varchar[]" : "varchar").Or.EqualTo(isArray ? "_text" : "text"), "#1.3");
```

Comment:
This is too fragile.. avoid using 'OR'.. verify the column/index is exactly what we expect for on each case. (And think this tests is going to be run under postgres, but also under sqlite, with different 'Features' enabled on each case).  So probably this test should have different set of Asserts depending on the actual Features enabled on this test run.

Reply:
As said, we should not have Asserts having 'variable' results. If the db & persister under tests has X feature enabled, the asserts *should* ensure just that. So remove 'Or' conditions and base them on specific enabled features.